library/item/extension/src/main/java/org/quiltmc/qsl/item/extension/api/ProjectileModifyingBowItem.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.item.extension.api;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.projectile.PersistentProjectileEntity;
+import net.minecraft.item.BowItem;
+import net.minecraft.item.ItemStack;
+import org.quiltmc.qsl.item.extension.impl.BowExtensions;
+
+/**
+ * This is the default implementation for {@link BowExtensions}, allowing for the easy creation of new bows with no new modded functionality.
+ * <p>
+ * This bow automatically registers itself to modify its arrows with {@link ProjectileModifyingBowItem#onProjectileShot(ItemStack, ItemStack, LivingEntity, float, PersistentProjectileEntity)}</p>
+ */
+public class ProjectileModifyingBowItem extends BowItem implements ShotProjectileEvents.ModifyProjectileFromBow {
```

Comment:
Since `extension` will encapsulate many different item types (bows, tridents, elytras, etc) I wonder if it would be better to organize sub-packages for `bow`, `trident`, `elytra`, etc.